modules/home-manager/globals.nix

```diff
@@ -11,5 +20,22 @@
       default = "teal";
       description = "Global Catppuccin accent";
     };
+
+    enable = lib.mkEnableOption "catppuccin globally";
   };
+
+  config.programs = lib.mkIf cfg.enable (
+    lib.foldlAttrs
+      (
+        programs: name: value:
+          programs
+          // (lib.optionalAttrs (lib.hasAttr "catppuccin" value) {
+            ${name}.catppuccin = {
+              enable = lib.mkDefault cfg.enable;
+            };
+          })
+      )
+      { }
+      options.programs
```

Comment:
i don't think this is the best way to go about this. the biggest issue here is the assumption that all the options for each port are in `options.programs` when this isn't the case - mako and dunst's options are under `services`, for example. i would also considered it a bit of a forced approach

imo the best solution would be editing `lib.ctp.mkCatppuccinOpt` to make the `enable` and `flavour` option shared by all modules default to `config.catppuccin.enable`/`config.catppuccin.flavour`

Reply:
Fair enough... I'll see what I can do.

Reply:
Also just realised this does not account for `delta` which is under `programs.git.delta`

Reply:
Would this be a better option ?